### PR TITLE
build(devs-infra): configure eslint

### PR DESCRIPTION
**Pull request by ahnpnl**
## Summary

- Configure eslint based on recommended **eslint/typescript-eslint** config and disable a few rules which are not necessary.

- Upgrade typescript to 3.9.7 which requires our test snapshots to be updated.

- Update `README.md` because the restructure PR didn't update it.

## Test plan

Green CI

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

**N.A.**

### CODE REVIEW

File: src/InlineFilesTransformer.ts [hunk 1]
```diff
@@ -81,14 +81,8 @@ export function factory(cs: ConfigSet) {
...
-  function isPropertyAssignmentToTransform(
-    node: Node
-  ): node is PropertyAssignment {
-    return (
-      ts.isPropertyAssignment(node) &&
-      ts.isIdentifier((node as PropertyAssignment).name) &&
-      TRANSFORM_PROPS.includes(((node as PropertyAssignment).name as Identifier).text)
-    );
+  function isPropertyAssignmentToTransform(node: Node): node is PropertyAssignment {
+    return ts.isPropertyAssignment(node) && ts.isIdentifier(node.name) && TRANSFORM_PROPS.includes(node.name.text);
```
Comment (wtho) [inline src/InlineFilesTransformer.ts:85]: Interesting, I was getting errors without the casts

**Review by wtho — APPROVED** [review]
Nice! Enhances code style consistency!

File: src/InlineFilesTransformer.ts (hunk 1 above)
Comment (ahnpnl) [reply src/InlineFilesTransformer.ts:85]: weird, I tested with both windows and macos, `yarn build` is fine for me. Maybe you need to remove `node_modules` and reinstall 😃

Comment (wtho) [reply src/InlineFilesTransformer.ts:85]: Maybe it was still on an older TS version with less inference when I originally implemented this.

**Pull request merged**